13167 docs for connector builder new UI schemaform rollout (#63329)
airbytehq/airbyte: docs/platform/connector-development/connector-builder-ui/incremental-sync.md

## Configuration

To configure incremental syncs for a stream in the connector builder, you need to specify the **cursor field**, **datetime formats**, **time filtering capabilities**, and **request injection** settings.

In the builder UI, these are configured as follows:

- **Cursor field** - The property in the record that defines when the record was last updated. This field is used to determine which records have been synced and which are new.
- **Cursor datetime formats** - The possible formats for the cursor field, in order of preference. The first format that matches the cursor field value will be used to parse it. The UI can auto-detect common formats from your test data.

> - **API time filtering capabilities** - Specifies how the API allows filtering by datetime:
>   - **Range**: API supports both start and end datetime filtering
>   - **Start**: API supports start datetime filtering but returns all data from start to now
>   - **No filter (data feed)**: API doesn't support filtering and returns data in descending order (newest to oldest)

Reviewer: ```suggestion
```
This went away in the transition to SchemaForm

As this fulfills the requirements for incremental syncs, we can configure the "Incremental sync" section in the following way:

- "Cursor field" is set to `webPublicationDate`
- "Cursor datetime formats" is set to `%Y-%m-%dT%H:%M:%SZ`

> - "API time filtering capabilities" is set to "Range" (supports both start and end datetime filtering)

Reviewer: ```suggestion
```

> - "Start datetime" is set to "user input" to allow the user of the connector configuring a Source to specify the time to start syncing

Reviewer: ```suggestion
- "Start datetime" is set to "Interpolated Value" and pointed at `{{ config['start_time'] }}` to allow the user of the connector configuring a Source to specify the time to start syncing
```

> - "End datetime" is set to "now" to fetch all articles up to the current date

Reviewer: ```suggestion
- "End datetime" is set to `{{ now_utc().strftime('%Y-%m-%dT%H:%M:%SZ') }}` to fetch all articles up to the current date
```

In some cases, it's helpful to reference the start and end date of the interval that's currently synced, for example if it needs to be injected into the URL path of the current stream. In these cases it can be referenced using the `{{ stream_interval.start_time }}` and `{{ stream_interval.end_time }}` [placeholders](/platform/connector-development/config-based/understanding-the-yaml-file/reference#variables). Check out [the tutorial](./tutorial.mdx#adding-incremental-reads) for such a case.
:::

> ## API time filtering capabilities

Reviewer: This section should be removed as this went away with the switch to SchemaForm.

Now, data feed is a toggle in the Advanced section called `Whether the target API is formatted as a data feed`

## Advanced settings

### Outgoing datetime format

The "Outgoing datetime format" specifies how datetime values are formatted when sent to the API in requests. If not specified, the first format from your cursor datetime formats will be used.

Supported format placeholders include:
- `%s` - Unix timestamp (seconds) - `1686218963`
- `%s_as_float` - Unix timestamp as float with microsecond precision - `1686218963.123456`
- `%ms` - Unix timestamp (milliseconds) - `1686218963123`
- `%Y-%m-%dT%H:%M:%S.%f%z` - ISO 8601 with microseconds and timezone
- `%Y-%m-%d` - Date only format
- And many other standard datetime format codes

For a complete list of format codes, see the [Python strftime documentation](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes).

The description above is sufficient for a lot of APIs. However there are some more subtle configurations which sometimes become relevant.

> ### Split up interval

Reviewer: This section should be rewritten, as `Split up interval` is now replaced with the `Cursor Granularity` and `Step` fields in Incremental sync (i.e. the fields that were inside of `Split up interval` before)
